### ISSUE TITLE
Tiedtke updraft: organized entrainment/detrainment is not in the metre-based, capped form

### ISSUE THREAD

**Issue by duncanwp** [climate-analytics-lab/jax-gcm#669]
Found by a re-audit of the July 2026 ECHAM+RRTMGP review against `dev` @ `f6d1bcd`. Parent: #614.

Both reference schemes express organized entrainment and detrainment as a **per-metre fractional rate with a hard cap of the same order** — ECHAM `centrmax = 3.0e-4 m⁻¹`, CAM ZM `2.0e-4 m⁻¹`. jcm uses neither form and caps neither quantity.

## 1. Organized detrainment is measured in level counts and never clipped

`updraft.py:308-323`:
```python
cloud_depth   = jnp.maximum(kbase - ktop, 1.0)          # LEVEL COUNT
frac_height   = jnp.clip((kbase - k) / cloud_depth, 0.0, 1.0)
org_profile   = jnp.maximum(jnp.tan(jnp.pi*(0.75*frac_height - 0.25)), 0.0)
detr_strength = 0.003 * jnp.sqrt(jnp.maximum(cloud_depth / 10.0, 1e-30))
detr          = detr_turb + detr_org                     # NOT clipped
```

ECHAM (`mo_cuascent.f90:780-784`) uses metres and scales as 1/depth:
```fortran
ztmzk  = -(pgeoh(jl,ikh)-pgeoh(jl,ikt))*zrg     ! plume depth in METRES
zorgde = TAN(zarg)*3.1415_wp*0.5_wp/ztmzk       ! prefactor ~ 1/depth
podetr(jl,kk) = MIN(zorgde,centrmax)*pmfu(jl,kk+1)*zdprho(jl)
```

Three consequences, all verified:

- **Resolution dependence with the wrong sign.** `0.003·sqrt(cloud_depth/10)` *increases* with level count (2.68e-3 /m at 8 levels, 5.20e-3 /m at 30 levels for the same physical cloud) where both references *decrease* it with physical depth.
- **No cap.** Measured on the standard tropical sounding, `detr·dz` per layer runs **19.2, 9.7, 6.5, 4.8, 3.7, 2.9, 2.3, 1.8, 1.35, 0.92 …** — up to 1900 % of the plume mass detrained in one layer, against ceilings of `centrmax·dz = 0.15` (ECHAM) and `2e-4·dz = 0.10` (ZM).
- **Detrainment, not buoyancy, sets the cloud top.** In the same probe the plume is annihilated at k=36, giving a 406 hPa cloud top on a sounding whose target ceiling is 150 hPa. And because `plude_layer = carry.lu[next_level] * dmf_detr` (`:425`) is not limited by available plume mass, that layer detrains condensate on **118 %** of the mass that ever entered it.

## 2. Nordeng organized entrainment omits `zdrodz` and caps the wrong quantity

`updraft.py:288-301`:
```python
entr_org = w_deep_in * zbuoyz * 0.5 / (1.0 + zbuoy_accum)
entr     = jnp.clip(entr_turb + entr_org, 0.0, 0.01)
```

ECHAM (`mo_cuascent.f90:520-526`):
```fortran
zdrodz = -LOG(pten(jk-1)/pten(jk))/zdz - grav/(rd*ptenh(jk)*(1+vtmpc1*pqenh(jk)))
zoentr(jk-1) = zbuoyz*0.5_wp/(1._wp+zbuoy(jl)) + zdrodz
zoentr(jk-1) = MIN(zoentr(jk-1),centrmax)
zoentr(jk-1) = MAX(zoentr(jk-1),0._wp)
```

- `zdrodz ≈ −1.0e-4 m⁻¹` in the lower troposphere — **comparable in magnitude to `entrpen = 1e-4` itself.** Omitting it leaves organized entrainment systematically too large and removes the `MAX(...,0)` mechanism by which ECHAM clamps entrainment to zero in weakly buoyant layers.
- ECHAM caps `zoentr` **alone** at `centrmax`; ZM likewise caps the entrainment *rate* at `2e-4` (`zm_convr.F90:1958`). jcm caps the **sum** at `0.01` — 33× ECHAM and 50× ZM. At dz = 500 m that permits 500 % entrainment in one layer where both references permit ≤ 15 %.

Measured on the tropical sounding: `state.entr` reaches 6.56e-4 m⁻¹ (2.2× `centrmax`), per-layer `entr·dz` runs 0.15–0.58, and plume mass flux **triples** from 0.94 at cloud base to 2.43 kg m⁻² s⁻¹ by 496 hPa.

## Why it matters

Together these are why deep plumes die mid-troposphere: entrained too fast, torn apart by unbounded organized detrainment, reaching roughly half their target depth — and the cloud top moves when you change vertical resolution at fixed physics.

## Fix

- Carry geopotential heights into the scan. Rebuild `detr_org` in metres: `ztmzk = z(kctop0) − z(khmin)`, `zzmzk = z(k) − z(khmin)`, `zorgde = tan(π·zzmzk/(2·ztmzk))·π/(2·ztmzk)`, clipped with a new `centrmax = 3.0e-4` parameter.
- Add `zdrodz = -log(T[k-1]/T[k])/dz - grav/(rd*T_half*(1+vtmpc1*q_half))` to `entr_org`, then apply ECHAM's clamp order: `clip(zbuoyz*0.5/(1+zbuoy_accum) + zdrodz, 0.0, centrmax)`. Drop the `0.01` clip on the sum.
- Cap detrained mass at `0.75·mfu[next_level]` before forming both `mfu_new` and `plude_layer`, so the plude ledger can never exceed the plume mass.

## Also here: the dead `cmfctop` knob

`cmfctop` is defined (`types.py:58, 86, 108`), defaulted to 0.2, Hydra-settable and documented, but referenced **nowhere** in the physics — the only non-test hit is an archaeology comment at `downdraft.py:394`. ECHAM uses it for the cloud-top overshoot layer (`mo_cuascent.f90:546-556`).

**The physics gap is defensible and should not be "fixed" reflexively:** CAM's ZM also detrains the entire residual plume at a single level with no overshoot (`zm_convr.F90:2074-2087`), and jcm's smooth `survival` sigmoid (`updraft.py:441-455`) spreads the dump over a few levels, which is gentler than ZM. So either wire the ECHAM overshoot step or **delete the parameter** — but do not leave a phantom knob that a calibration sweep silently ignores.

## Provenance

Checked against ECHAM 6.3-HAM2.3 and ESCOMP/CAM `cam6_4_196` (ZM via `atmos_phys0_27_000`). Nothing in ZM, `uwshcu` or `convect_deep` measures entrainment or detrainment in level counts; the level-count form is a jcm invention with no provenance in either reference.

## Refs

Review findings 2.6b, 2.8f, 2.6c. Extends CONV-2 from the retired `REMAINING_ISSUES.md`.
